indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -412,6 +430,63 @@ private TaskStatus runInternal(TaskToolbox toolbox) throws Exception
     }
   }
 
+  @Override
+  public boolean canRestore()
```

Comment:
Rather than tying the MR job cleanup to the restore functionality, I think it would be better to change the `Task.stopGracefully()` contract such that it's always called.

Regardless of whether the task would be restored, I think it makes sense to call `stopGracefully()` in case the task wants to attempt to clean up any open resources, like in this situation.

Also, as is, the MR job termination won't run unless the user has enabled `restoreTasksOnRestart` which is false by default:

```
      if (taskConfig.isRestoreTasksOnRestart() && task.canRestore()) {
```